### PR TITLE
[E2E Test] Test time based upgrade is propogated correctly to full nodes

### LINKED ISSUE #480: [E2E Test] Test time based upgrade is propogated correctly to full nodes
### Test Case Type

New

### DRS

ALL

### Target Milestone

3D

### Test Case Issue link

_No response_

### Epic

time based upgrade

### Test Case Description

na

### Preconditions

1. One sequencer
2. One full node 
3. Sequencer produces block at 10 sec interval 

### Test Steps

1. Create a gov proposal to change rollapp params version (which is basically an upgrade - which should halt the chain) at specific time

### Expected Results

1. Sequencer is halted at max 10 sec from the upgrade time (with the assumption that there are txs executed)
2. Full node is halted at max 10 sec from upgrade time
3. After upgrade to new versions both full node and sequencer continue to operate normally 

### Live Test

- [ ] Yes

### Comments

_No response_

### PR DESCRIPTION
**Pull request by hungdinh82**
Close #480

### CODE REVIEW

**Comment by hungdinh82** [comment]
Close #480

File: tests/time_upgrade_test.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	dymintTomlOverrides["max_idle_time"] = "3s"
+	dymintTomlOverrides["max_proof_time"] = "500ms"
+	dymintTomlOverrides["batch_submit_time"] = "50s"
+	dymintTomlOverrides["p2p_blocksync_enabled"] = "false"
+
+	configFileOverrides["config/dymint.toml"] = dymintTomlOverrides
+	// Create chain factory with dymension
+	numHubVals := 1
+	numHubFullNodes := 1
+	numRollAppFn := 0
```
Comment (anhductn2001) [inline tests/time_upgrade_test.go:46]: Precondition requires 1 fullnode

File: tests/time_upgrade_test.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+	// Assuming block height is located in status["SyncInfo"]["latest_block_height"]
+	rollappHeightBeforeUpgrade, err := rollapp1.GetNode().Height(ctx)
+	require.NoError(t, err, "Failed to query Rollapp1 height before upgrade")
+	fmt.Printf("Rollapp1 current block height before upgrade version: %d\n", rollappHeightBeforeUpgrade)
+
+	//Wait 10s for create a gov proposal to change rollapp params version
+	time.Sleep(10 * time.Second)
+
+	// Change gov upgrade version
```
Comment (anhductn2001) [inline tests/time_upgrade_test.go:150]: It's not change param version proposal.

File: tests/time_upgrade_test.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+	for time.Since(startTime) < 10*time.Second {
+		_, err := rollapp1.GetNode().Height(ctx)
+		if err == nil {
+			chainIsBackOnline = true
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+	require.True(t, chainIsBackOnline, "Chain did not come back online within 10 seconds after upgrade")
```
Comment (anhductn2001) [inline tests/time_upgrade_test.go:183]: This condition isn't true

**Review by anhductn2001 — APPROVED** [review]

**Pull request merged**